### PR TITLE
Additional methods + fix remove_cog

### PR DESCRIPTION
**Pull request by chillymosh**
Added fetch_follow to client method to compare any 2 users relationship (from 1.2.3 missing from 2.0)
Can use ID or name

Added fetch_channel to client
You can now fetch the data for any channel without using private _http private methods using ID or name

I have a fix for remove_cog in my PR which now makes it work when called directly or through unload_module.
It worked fine if you call it directly e.g. self.remove_cog('Cogname') but when you try and unload a module it failed since it was a string and not a cog object. Adding a try: except: resolves it.

Tided up some general code

### CODE REVIEW

File: twitchio/parse.py [hunk 1]
```diff
@@ -51,7 +47,7 @@ def parser(data: str, nick: str):
     if action == "PING":
         return dict(action="PING")
 
-    elif groups[2] == "PRIVMSG" or groups[2] == "PRIVMSG(ECHO)":
+    elif groups[2] in ["PRIVMSG", "PRIVMSG(ECHO)"]:
```
Comment (IAmTomahawkx) [inline twitchio/parse.py:50]: Maybe a set here

**Review by IAmTomahawkx — COMMENTED** [review]
I'm not sure if you've used the proper formatting on this, we use black with a *120* line length

File: twitchio/parse.py (hunk 1 above)
Comment (chillymosh) [reply twitchio/parse.py]: Changed to set

File: twitchio/client.py [hunk 2]
```diff
@@ -287,6 +287,26 @@ def get_channel(self, name: str) -> Optional[Channel]:
...
+        --------
+            :class:`list`
+        """
+
+        if not broadcaster.isdigit():
+            get_id : User = await self.fetch_users(names=[broadcaster.lower()])
+            for i in get_id:
+                broadcaster = i.id
+
+        return await self._http.get_channels(broadcaster)
```
[the remark this answers is not in the document]
Comment (chillymosh) [reply twitchio/client.py:322]: Created a new model called ChannelInfo and this now returns that

<ChannelInfo user=<PartialUser id=21734222, name=chillymosh> game_id=26936 game_name=Music title=Testing New  language=en delay=0>

File: twitchio/client.py [hunk 3]
```diff
@@ -419,10 +455,43 @@ async def fetch_videos(
...
-            ids, user_id=user_id, game_id=game_id, period=period, sort=sort, type=type, language=language
+            ids,
+            user_id=user_id,
+            game_id=game_id,
+            period=period,
+            sort=sort,
+            type=type,
+            language=language,
         )
         return [Video(self._http, x) for x in data]
```
[the remark this answers is not in the document]
Comment (chillymosh) [reply twitchio/client.py:496]: Moved this to user.py from client and is access as PartialUser.fetch_follow(PartialUser)  can use to_user kwarg and token= kwag
returns
<FollowEvent from_user=<PartialUser id=21734222, name=chillymosh> to_user=<PartialUser id=247716086, name=McRocklin> followed_at=2020-09-10 21:05:59>

**Comment by chillymosh** [comment]
Changed my black line settings to 120, I had missed that setting on the workspace I was using.

File: twitchio/ext/commands/bot.py [hunk 4]
```diff
@@ -370,16 +368,18 @@ def add_cog(self, cog: Cog):
...
             The name of the cog to remove.
         """
-        if cog_name not in self._cogs:
-            raise InvalidCog(f"Cog '{cog_name}' not found")
+        try:
+            if cog_name not in self._cogs:
+                raise InvalidCog(f"Cog '{cog_name}' not found")
 
-        cog = self._cogs.pop(cog_name)
+            cog = self._cogs.pop(cog_name)
```
[the remark this answers is not in the document]
Comment (chillymosh) [reply twitchio/ext/commands/bot.py:380]: I'm digging deeper into this one as there is something else at play here

File: twitchio/user.py [hunk 5]
```diff
@@ -413,6 +412,30 @@ async def fetch_followers(self, token: str = None):
         data = await self._http.get_user_follows(to_id=str(self.id))
         return [FollowEvent(self._http, d, to=self) for d in data]
 
+    async def fetch_follow(self, to_user, token: str = None):
```
Comment (IAmTomahawkx) [inline twitchio/user.py:415]: to_user isnt typed

**Review by IAmTomahawkx — CHANGES_REQUESTED** [review]

File: twitchio/user.py (hunk 5 above)
Comment (chillymosh) [reply twitchio/user.py]: This is now type hinted in latest push

**Comment by chillymosh** [comment]
Changes have been made

**Review by IAmTomahawkx — APPROVED** [review]

**Pull request merged**